swupd/packs_test.go

```diff
@@ -637,22 +687,46 @@ func TestPackRenames(t *testing.T) {
 	mustHaveDeltaCount(t, info, 1)
 	checkFileInPack(t, ts.path("www/20/pack-os-core-from-10.tar"), fmt.Sprintf("delta/10-20-%s-%s", hashIn10, hashIn20))
 
+	// Check Delta Manifest 10->20
+	{
+		m := checkDeltaManifest(ts, 10, 20, "os-core", 1)
+		fileInManifestHash(t, m, 10, "/file1", hashIn10)
+	}
+
 	// Pack from 20->30 will contain a delta due to content change (and rename).
 	info = ts.createPack("os-core", 20, 30, ts.path("image"))
 	mustHaveDeltaCount(t, info, 1)
 	checkFileInPack(t, ts.path("www/30/pack-os-core-from-20.tar"), fmt.Sprintf("delta/20-30-%s-%s", hashIn20, hashIn30))
 
+	// Check Delta Manifest 20->30
+	{
+		m := checkDeltaManifest(ts, 20, 30, "os-core", 1)
+		fileInManifestHash(t, m, 20, "/file1", hashIn20)
+	}
+
 	// Pack from 10->30 will contain a delta due to content change (and rename).
 	info = ts.createPack("os-core", 10, 30, ts.path("image"))
 	mustHaveDeltaCount(t, info, 1)
 	checkFileInPack(t, ts.path("www/30/pack-os-core-from-10.tar"), fmt.Sprintf("delta/10-30-%s-%s", hashIn10, hashIn30))
 
+	// Check Delta Manifest 10->30
+	{
+		m := checkDeltaManifest(ts, 10, 30, "os-core", 1)
+		fileInManifestHash(t, m, 10, "/file1", hashIn10)
+	}
+
 	// Pack from 10->40 will contain a delta due to content change (and rename).
 	info = ts.createPack("os-core", 10, 40, ts.path("image"))
 	mustHaveDeltaCount(t, info, 1)
 
 	// Note that the delta refers to the version of the file, which is still 30.
 	checkFileInPack(t, ts.path("www/40/pack-os-core-from-10.tar"), fmt.Sprintf("delta/10-30-%s-%s", hashIn10, hashIn30))
+
+	// Check Delta Manifest 10->40
+	{
+		m := checkDeltaManifest(ts, 10, 40, "os-core", 1)
+		fileInManifestHash(t, m, 10, "/file1", hashIn10)
+	}
```

Comment:
These are good tests, can you please add a few negative cases as well? We want to make sure the delta manifest contains only those files that were used as from deltas and no other files.

Reply:
True. I'll check the number of files in the manifest.